### PR TITLE
Adding advanced documentation

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Closes: #684 

Changes proposed in this PR:

- Create new documentation folder
- Create a .md file for advance features/ settings
- Document RBAC server
- Document Free pricing
- Document allow / deny lists in the DDO

### CODE REVIEW

File: docs/advanceDocumentation.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- [Free Pricing](#-free-pricing-settings)
+
+## RBAC settings
+
+- Setup and host the Ocean role based access control (RBAC) server. Follow the instructions in the RBAC repository: https://github.com/oceanprotocol/RBAC-Server
+- The RBAC server can store roles in [Keycloak](https://www.keycloak.org/) or a json file.
+- Create an environmental variable in the .env file called `GATSBY_RBAC_URL`. The value should by the URL of the Ocean RBAC server that you have hosted, e.g. `GATSBY_RBAC_URL= "http://localhost:3000"`
+- Users of your marketplace will now require the correct role ("user", "comsumer", "publisher") to access features in your marketplace. The market will check the role that has been allocated to the user based on the address that they have connected to the market with.
+
+## Allow Deny List Settings
```
Comment (mihaisc) [inline docs/advanceDocumentation.md:16]: Allow/Deny (or Allow and Deny) seems better, Allow Deny might be misunderstood

**Comment by jamiehewitt15** [comment]
@mihaisc Thank you for the review, I have integrated your feedback

File: README.md [hunk 2]
```diff
@@ -374,6 +375,16 @@ Additionally, we would also advise that your retain the text saying "Powered by
 
 Everything else is made open according to the apache2 license. We look forward to seeing your data marketplace!
 
+## Advanced Features
+
+Ocean Market also includes a number of advance features that are suitable for an enterprise data market, such as:
```
Comment (kremalicious) [inline README.md:380]:
```suggestion
Ocean Market also includes a number of advanced features that are suitable for an enterprise data market, such as:
```

File: README.md [hunk 3]
```diff
@@ -374,6 +375,16 @@ Additionally, we would also advise that your retain the text saying "Powered by
...
+
+- Role based access control
+- Allow and deny lists
+- Free pricing
+
+[See our seperate guide on advance features](docs/advanceDocumentation.md)
```
Comment (kremalicious) [inline README.md:386]:
```suggestion
[See our seperate guide on advanced features](docs/advancedDocumentation.md)
```

so also change the file name to `docs/advancedDocumentation.md`. And maybe rather call it `docs/advanced-settings.md`?

File: docs/advanceDocumentation.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Advance Settings
```
Comment (kremalicious) [inline docs/advanceDocumentation.md:1]:
```suggestion
# Advanced Settings
```

File: docs/advanceDocumentation.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+**Table of Contents**
+
+- [Role based Access Control](#-rbac-settings)
+- [Allow and Deny lists](#-allow--and-deny-list-settings)
+- [Free Pricing](#-free-pricing-settings)
+
+## RBAC settings
+
+- Setup and host the Ocean role based access control (RBAC) server. Follow the instructions in the RBAC repository: https://github.com/oceanprotocol/RBAC-Server
```
Comment (kremalicious) [inline docs/advanceDocumentation.md:11]: better to have no URLs as cleartext but always link them up. In this case you could link up e.g. `role based access control (RBAC) server` or `RBAC repository`

File: docs/advanceDocumentation.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+- The RBAC server can store roles in [Keycloak](https://www.keycloak.org/) or a json file.
+- Create an environmental variable in the .env file called `GATSBY_RBAC_URL`. The value should be the URL of the Ocean RBAC server that you have hosted, e.g. `GATSBY_RBAC_URL= "http://localhost:3000"`
```
Comment (kremalicious) [inline docs/advanceDocumentation.md:13]: wee can assume people have already copied `.env.example` to `.env` as thee main readme outlines. So people have that `GATSBY_RBAC_URL` already there so we should modify these instructions here a bit

File: docs/advanceDocumentation.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+- Users of your marketplace will now require the correct role ("user", "consumer", "publisher") to access features in your marketplace. The market will check the role that has been allocated to the user based on the address that they have connected to the market with.
+
```
Comment (kremalicious) [inline docs/advanceDocumentation.md:15]: would love to see another sentence on how this works within the app, so mentioning the `Permission` component

**Comment by jamiehewitt15** [comment]
@kremalicious thanks a lot for the feedback, I have integrated your suggestions.

**Review by kremalicious — APPROVED** [review]

**Pull request merged**